Core/GDCore/Project/Project.cpp

```diff
@@ -642,6 +683,29 @@ void Project::UnserializeFrom(const SerializerElement& element) {
   if (currentPlatform == NULL && !platforms.empty())
     currentPlatform = platforms.back();
 
+  // TODO EBO All events based behaviors and events based objects should be
+  // added to the project before the content of events based objects is
+  // unserialized because when unserializing child-objects InsertNewObject and

```

Comment:
Might be a good reason for storing separately the data of an object (something like "gd::BehaviorContent") from the code manipulating it (something like "gd::Behavior") (noting that the naming is not perfect)